Refactor and small fix for the cyborg power connector

Sarchutar opened the pull request:
## About The Pull Request

Fixes #7283

Refactors the cyborg power connector to be a bit more readable.
Changes the `attack_self` message to a balloon alert. 
Makes it use a progress bar in `do_after`  (why did it not use one initially?).
Addition of a small limit for when charging things, so cyborgs don't completely drain their cells.

Refactor also fixes a bug that made the connector unable to be used after moving while draining power from a power cell.


## Why It's Good For The Game

More readable stuff is always good i guess, the balloon alert/progress bar is nice to have and the limit is just a safety for charging things.
And as a side thing, bug fixes are good.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/53494785/183272855-3c1b7e15-d0db-4297-b6ba-c97c781f9168.mp4

</details>

## Changelog
:cl:
refactor: refactored the cyborg power connector code
add: changing modes of the power connector now uses a balloon alert
add: limit when charging things to not accidentally completely drain ones cell
add: when drawing power/charging it has a progress bar now
fix: fixed a bug which made the connector unusable when moving while drawing power from a cell
/:cl:

Sarchutar committed:
refactoring of the power connector

- refactoring of the cyborg power connector
- refactor also fixes connector never changing active back to false when moving while drawing from cells

DrDuckedGoose commented on code/game/objects/items/robot/robot_items.dm:
```diff
@@ -220,77 +237,123 @@
+	while(do_after(user, 15, target = target, extra_checks = CALLBACK(src, .proc/mode_check)))
+		if(!user || !user.cell)
+			active = FALSE
+			return
```
This is a completely optional suggestion, this PR is pretty flawless.
The original code here is more readable than this suggestion.
```
		if(!user?.cell)
			active = FALSE
			return
```

Sarchutar committed:
review things

- changes if(!user || !user.cell) to if(!user?.cell)

Co-Authored-By: DrDuckedGoose <40559528+DrDuckedGoose@users.noreply.github.com>

Sarchutar committed:
improve the limiter

- improves the limiter by subtracting the minimum charge cyborgs should have left from the cyborgs current charge in the calculation of how much charge should be used/given to objects
